src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -1747,21 +1748,34 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         case NI_AdvSimd_StoreVector64x2:
         case NI_AdvSimd_Arm64_StoreVector128x2:
         {
-            assert(sig->numArgs == 2);
             assert(retType == TYP_VOID);
+            assert(sig->numArgs == 2);
 
             CORINFO_ARG_LIST_HANDLE arg1     = sig->args;
             CORINFO_ARG_LIST_HANDLE arg2     = info.compCompHnd->getArgNext(arg1);
             var_types               argType  = TYP_UNKNOWN;
             CORINFO_CLASS_HANDLE    argClass = NO_CLASS_HANDLE;
-
             argType             = JITtype2varType(strip(info.compCompHnd->getArgType(sig, arg2, &argClass)));
             op2                 = impPopStack().val;
             unsigned fieldCount = info.compCompHnd->getClassNumInstanceFields(argClass);
-            argType             = JITtype2varType(strip(info.compCompHnd->getArgType(sig, arg1, &argClass)));
-            op1                 = getArgForHWIntrinsic(argType, argClass);
 
-            assert(op2->TypeGet() == TYP_STRUCT);
+            if (op2->TypeGet() == TYP_STRUCT)
```

Comment:
I don't think you need this check since this is just for `StoreVector64x2` and `StoreVector128x2`. This case should be untouched now.

Reply:
True... rolled back all the changes to StoreVectorNx2.